controllers/machinehealthcheck_controller_test.go

```diff
@@ -1460,3 +1382,232 @@ func ownerReferenceForCluster(ctx context.Context, c *clusterv1.Cluster) metav1.
 		UID:        cc.UID,
 	}
 }
+
+func setup(g *WithT) (*clusterv1.Cluster, *clusterv1.MachineHealthCheck) {
```

Comment:
```suggestion
func createNamaspaceClusterAndMHC(g *WithT) (*clusterv1.Cluster, *clusterv1.MachineHealthCheck) {
```
I find kind of confusing this func creating ns and cluster, while returning the MHC without creating int.what about adding an an input parameter providing a func to mutate the mhc, so we can create it inside this func?